src/can.c

```diff
@@ -109,8 +170,39 @@ static void thingset_can_report_rx_cb(const struct device *dev, struct can_frame
     const struct thingset_can *ts_can = user_data;
     uint16_t data_id = THINGSET_CAN_DATA_ID_GET(frame->id);
     uint8_t source_addr = THINGSET_CAN_SOURCE_GET(frame->id);
-
-    ts_can->report_rx_cb(data_id, frame->data, can_dlc_to_bytes(frame->dlc), source_addr);
+#ifdef CONFIG_THINGSET_CAN_PACKETIZED_REPORTS_RX
+    if (THINGSET_CAN_PACKETIZED_REPORT(frame->id)) {
+        struct net_buf *buffer = NULL;
+        if ((buffer = thingset_can_get_rx_buf(source_addr)) != NULL) {
+            struct thingset_can_rx_context *context = (struct thingset_can_rx_context *)buffer->user_data;
```

Comment:
Where is this user_data set? Not sure I understand this correctly.

Reply:
Lines 82-87:

```c
buffer = net_buf_alloc(&thingset_can_rx_buffer_pool, K_NO_WAIT);
    if (buffer != NULL) {
        struct thingset_can_rx_context *context = (struct thingset_can_rx_context *)buffer->user_data;
        context->src_addr = src_addr;
        context->seq = 0;
        context->escape = false;
```